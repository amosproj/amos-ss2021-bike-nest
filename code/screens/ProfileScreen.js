import React, { Component } from "react";
import { ScrollView, StyleSheet, Text, View, Image, Pressable } from "react-native";
import { mainStyles } from "../styles/MainStyles";

// This is the screen where the Profile is shown and additional information can be seen

export default function ProfileScreen({ navigation }) {
        return (
            <View style={mainStyles.container}>
                <ScrollView>
                    <Text style={mainStyles.h1}>Profil</Text>
                    <View style={myStyles.profile}>
                        <Image source={require('../assets/Avatar.png')}></Image>
                        <View style={mainStyles.profileInfo}>
                            <Text style={mainStyles.h3}>
                                Max Muster {"\n"} 
                            </Text>
                            <Text style={mainStyles.stdText}>
                                Beispiel@Gmail.com {"\n"} 
                                +52 1111111111 {"\n"} 
                                Adresse
                                {"\n"} 
                                {"\n"} 
                            </Text>
                            <Pressable onPress={()=>navigation.navigate("EditPersonalInformation")} >
                                <View style={mainStyles.buttonContent}>
                                    <Image style={[myStyles.buttonImage, { maxWidth: 150, resizeMode: 'contain' }]} source={require('../assets/edit-3.png')}></Image>
                                </View>
                            </Pressable>
                            
                        </View >
                    </View>
                </ScrollView>
            </View>
        )
    }

const myStyles = StyleSheet.create({
    profile:{
        flexDirection: 'row',
        margin: 10,
    },
    profileInfo:{
        flexDirection: 'column'
    }
})
