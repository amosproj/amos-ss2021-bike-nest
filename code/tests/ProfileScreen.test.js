import React from 'react';
import renderer from 'react-test-renderer';
import ProfileScreen from '../screens/ProfileScreen';

it('renders correctly', () => {
    const tree = renderer
        .create(<ProfileScreen />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});