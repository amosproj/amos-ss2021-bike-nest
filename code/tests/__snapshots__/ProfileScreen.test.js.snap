// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 28,
              "fontWeight": "bold",
              "margin": 30,
              "padding": 15,
            },
            Object {
              "alignSelf": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        Profil
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "margin": 10,
          }
        }
      >
        <Image
          source={1}
        />
        <View>
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "300",
              }
            }
          >
            Max Muster 
            

          </Text>
          <Text
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            Beispiel@Gmail.com 
            

            +52 1111111111 
            

            Adresse
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "justifyContent": "flex-end",
            "marginBottom": 20,
            "marginRight": 20,
            "marginTop": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View>
            <Image
              source={1}
              style={
                Array [
                  undefined,
                  Object {
                    "maxWidth": 150,
                    "resizeMode": "contain",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 38,
            "elevation": 3,
            "height": 55,
            "justifyContent": "center",
            "margin": 9,
            "opacity": 1,
            "padding": 18,
            "width": 328,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "color": "#000000",
                },
              ]
            }
          >
            Zahlungsinformationen
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 38,
            "elevation": 3,
            "height": 55,
            "justifyContent": "center",
            "margin": 9,
            "opacity": 1,
            "padding": 18,
            "width": 328,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "color": "#000000",
                },
              ]
            }
          >
            Ãœber Bike Nest
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 38,
            "elevation": 3,
            "height": 55,
            "justifyContent": "center",
            "margin": 9,
            "opacity": 1,
            "padding": 18,
            "width": 328,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "color": "#000000",
                },
              ]
            }
          >
            FAQ
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 38,
            "elevation": 3,
            "height": 55,
            "justifyContent": "center",
            "margin": 9,
            "opacity": 1,
            "padding": 18,
            "width": 328,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "color": "#000000",
                },
              ]
            }
          >
            Hilfe
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "margin": 20,
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "textDecorationLine": "underline",
            }
          }
        >
          Log out
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Profil
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          (un)lock
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Navigation
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Verlauf
        </Text>
      </View>
    </View>
  </View>
</View>
`;
